### ISSUE TITLE
Use javascript based mdns rather than relying on system level integrations

### ISSUE THREAD

**Issue by ammmze** [mKeRix/room-assistant#205]
Before I go too much farther testing this out, I wanted to toss up what I'm working on here to see if it would valuable enough for a PR. I imagine in most scenarios the current mdns solution works just fine. But I've got a scenario where I'm wanting to spin up room-assistant in docker on macOS for some development work. In this scenario the discovery doesn't function. I've had a tiny bit of experience with this in another project I put together ([mashupmill/sonoff-diy-tools](https://github.com/mashupmill/sonoff-diy-tools)) but I had come across another library for DNS service discovery `dnssd2`. 

`dnssd2` is a pure js implementation, so it should be much more portable and the interface is quite similar to `mdns`. However, it doesn't have the resolver sequence thing that `mdns` has. Though I'm not sure if that is required or not since I'm not entirely sure what all it does. My testing so far has only been limited to running 2 docker instances side by side and confirming they see each other and one of them is elected as leader.

So if there is interest I can clean this up and submit a pull request.

https://github.com/ammmze/room-assistant/commit/d72a51746005e977b2f297e8cc9ac401205b9093

**Comment by mKeRix** [comment]
I skimmed over what you wrote, that looks good! Back then I made the choice to use the `mdns` library because I read online that the native version is more reliable and the original `dnssd` was abandoned. Since then the native dependencies were a bit annoying in multiple aspects though, so I would love to have a JS-only solution if it works just as well.

The resolver sequence is only defined there to solve some issue surrounding Raspbian and IPv4/IPv6 as well as address resolving within Docker. I assume that with the pure JS implementation you won't need it.

So yes, I'd certainly be interested in a PR. :)
